tests/migration_test.go

```diff
@@ -2178,17 +2191,20 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				})
 
 				It("[test_id:5004] should be migrated successfully, using guest agent on VM with postcopy", func() {
-					guestAgentMigrationTestFunc(v1.MigrationPostCopy, dv.Name, resource.MustParse("1Gi"))
+					guestAgentMigrationTestFunc(dv.Name, resource.MustParse("1Gi"), migrationPolicy)
 				})
 
 			})
 
-			It("[test_id:4747] should migrate using cluster level config for postcopy", func() {
```

Comment:
@iholder101 Do we still have a test covering the cluster config behavior? I would say we should use policies 99% of the time and just leave a few tests exercising  that we don't discard cluster config. WDYT?

Reply:
The idea sounds reasonable, but how to do suggest to execute it? Is there a way to express "run this 1% of the times?"

We can always check both cluster-wide and policy settings, but then we go back to having a non-parallel test which is what I tried to avoid in the first place.

WDYT?

Reply:
We can, inside the test, randomly choose one of the strategies with a probability of 1% to go with the cluster-wide setting, but I'm not sure it's acceptable.

cc @dhiller

Reply:
That clearly is non-deterministic behaviour which we do not want in a test. 

Maybe one serial test for testing a cluster-wide resource is acceptable if it is justified and the remaining ones are parallel.

Reply:
@dhiller In Kubernetes for example, the have two kinds of CI lanes, serial and non-serial.

The idea there is that the non-serial lane should run as much test as possible, and only if there's no choice the tests go into the serial lane.

It is very similar to our situation, except they change KubeletConfiguration and we change Kubevirt CR. Perhaps we should move into a similar model.

In addition, the serial lanes are more flakie by nature, so another mental benefit is that people try to make their tests as parallel as possible so they would end up in the non-serial lane.

Reply:
We are following the same practice, since we [advocate that only if we absolutely must run a test in serial, i.e. if global resources are modified, then the test should be serial](https://github.com/kubevirt/kubevirt/blob/c90227368b48a45180b93159515f6abf8a8af73e/tests/README.md?plain=1#L3).

ATM our presubmit lanes are running parallel tests first and then if and only if the parallel tests do not fail run the serial tests.

@iholder101 I am not sure whether we want to have separate lanes for serial and non serial. That would double all the lanes, which would double the overhead and double the number of jobs.

Maybe I am not getting what the advantage of that model over ours is, could you elaborate?

Reply:
Just to clarify, I meant tests, not test runs. Therefore we should leave the basic test which tests that a configuration from Kubevirt CR is still used if no policy applies.

Reply:
@dhiller Running the lanes separately will probably result in the parallel lanes finishing up much quicker, allowing a better feedback loop with the developer. However, I'm not sure that separating the lanes makes sense if there's so much overhead involved.

Maybe we can do something different - add a `test/serial-test-added` label to such PRs to let the reviewers know that a serial test is added. Maybe this way we could better bring this into the reviewer's attention? Just an idea.

Reply:
@xpivarc @dhiller I fear that going this path we might actually hurt CI instead of improving it, since for many tests that are now parallelized we would need to add more serial tests.

WDYT about adding a separate set of functional tests ensuring that migration policies take precedence of cluster-wide settings, and that if they don't exist the cluster-wide settings are used? This is already covered by [unit tests](https://github.com/kubevirt/kubevirt/blob/v1.0.0/pkg/virt-controller/watch/migration_test.go#L1589).

I think it might be better in the sense that we won't feel the need to keep more similar serial tests in the future. This post-copy test, for example, is complex and demands relatively a lot of resources. I think that simply testing that the cluster-wide settings result in the same migration configs would be easier and clearer.

WDYT?

Reply:
> @dhiller Running the lanes separately will probably result in the parallel lanes finishing up much quicker, allowing a better feedback loop with the developer. However, I'm not sure that separating the lanes makes sense if there's so much overhead involved.

Again, the parallel tests are running first, if they fail, the serial ones aren't even executed. If it was a separate lane, they would get executed regardless.

> 
> Maybe we can do something different - add a `test/serial-test-added` label to such PRs to let the reviewers know that a serial test is added. Maybe this way we could better bring this into the reviewer's attention? Just an idea.

Yeah, good idea.

Reply:
> @xpivarc @dhiller I fear that going this path we might actually hurt CI instead of improving it, since for many tests that are now parallelized we would need to add more serial tests.
> 
> WDYT about adding a separate set of functional tests ensuring that migration policies take precedence of cluster-wide settings, and that if they don't exist the cluster-wide settings are used? This is already covered by [unit tests](https://github.com/kubevirt/kubevirt/blob/v1.0.0/pkg/virt-controller/watch/migration_test.go#L1589).

If the unit tests are already covering the precedence scenario, why do we need an e2e test?

> 
> I think it might be better in the sense that we won't feel the need to keep more similar serial tests in the future. This post-copy test, for example, is complex and demands relatively a lot of resources. I think that simply testing that the cluster-wide settings result in the same migration configs would be easier and clearer.
> 
> WDYT?

Reply:
Good point :) @xpivarc WDYT?

Reply:
One serial test will not kill our CI. I would be for keeping one.

Reply:
Alright, if you insist :)
Now a serial test is also running.